test/steps/build.py

```diff
@@ -13,10 +13,22 @@ def step_impl(context, folder):
     os.chdir(parent_dir)
     assert os.path.isdir(folder), "Folder does not exist"
     os.chdir(folder)
+    
+@given('I create folder {folder}')
+def step_impl(context, folder):
+    os.makedirs(folder, exist_ok=True)
+    if not hasattr(context, 'created_folders'):
+        context.created_folders = []
+    context.created_folders.append(folder)
 
 @step('file {file} exists')
-def step_impl(context, file):
-    assert os.path.isfile(file), f"File {file} does not exist"
+@step('file {file} does {should} exist')
+def step_impl(context, file, should: str = ""):
+    should = (not "no" in should)
+    if should:
+        assert os.path.isfile(file), f"File {file} does not exist"
+    else:
+        assert not os.path.isfile(file), f"File {file} does exist"
```

Comment:
@Fubukimaru Correct me if I'm wrong, but aren't the messages in the `assert` defined backwards?
- `if should` --> The file should exists, then the assert returns `f"File {file} does exist"`
- `else` --> The file shouldn't exist, then the assert returns `f"File {file} does exist"`

Reply:
if the file `should` exist, check it does exist, otherwise throw `file does not exist`

Reply:
Thanks! I misread the code and thought that it was the message that the assert returned, not the exception